### PR TITLE
Do not prepare lr scheduler as it as the right number of steps

### LINKED ISSUE #23986: learn_rate behavior is not expected when using =transformers.TrainingArguments
### System Info

- `transformers` version: 4.30.0.dev0
- Platform: Linux-5.15.0-72-generic-x86_64-with-glibc2.31
- Python version: 3.9.7
- Huggingface_hub version: 0.14.1
- Safetensors version: 0.3.1
- PyTorch version (GPU?): 1.13.0+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

uniform_finetune.py:
``` python    
trainer = transformers.Trainer(
        model=model,
        train_dataset=train_data,
        eval_dataset=val_data,
        args=transformers.TrainingArguments(
            per_device_train_batch_size=args.per_gpu_train_batch_size,
            gradient_accumulation_steps=args.gradient_accumulation_steps,
            warmup_steps=warmup_steps,
            num_train_epochs=args.epochs,
            learning_rate=args.learning_rate,
            fp16=True,
            logging_steps=20,
            evaluation_strategy="steps",
            save_strategy="steps",
            eval_steps=saving_step ,
            save_steps=saving_step,
            output_dir=output_dir,
            save_total_limit=11,
            load_best_model_at_end=True if args.val_set_size > 0 else False,
            ddp_find_unused_parameters=False if ddp else None,
        ),
        data_collator=transformers.DataCollatorForSeq2Seq(tokenizer, return_tensors="pt", padding=True)
    )
```

when run `torchrun --nnodes 1 --nproc_per_node $ngpu uniform_finetune.py`  in 

### Expected behavior

before 
image

now 
image

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

Right now the LR scheduler is prepared by the Accelerate but it has a number of steps that already accounts for the number of processes. This results in the LR scheduler being stepp through num_processes too fast. This PR thus removes the lr_scheduler from the prepared objects.

Should fix #23986

### CODE REVIEW

**Review by muellerzr — APPROVED** [review]
Thanks for the fix!

**Review by pacman100 — APPROVED** [review]
Thank you @sgugger for the deep dive and super quick PR resolving the issue, LGTM! 🤗
